portfolio/src/main/java/com/google/sps/data/TimeRangeGroupArrayList.java

```diff
@@ -0,0 +1,121 @@
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/** Models an implementation of the TimeRangeGroup model using ArrayList. */
+public class TimeRangeGroupArrayList implements TimeRangeGroup {
+
+  public List<TimeRange> allTimeRanges;
```

Comment:
Consider adding some comment that indicates that at each moment this variable is already sorted.

Reply:
nit: I think you can make this `final` as well.

Reply:
Updated with comment and `final` declaration.